autoload/vital/__vital__/VS/Vim/Window/FloatingWindow.vim

```diff
@@ -237,38 +256,52 @@ endif
 " close
 "
 if has('nvim')
-  function! s:_close(win) abort
-    call nvim_win_close(a:win, v:true)
+  function! s:_close(winid) abort
+    call nvim_win_close(a:winid, v:true)
     call s:_notify_closed()
   endfunction
 else
-  function! s:_close(win) abort
-    call popup_close(a:win)
+  function! s:_close(winid) abort
+    call popup_close(a:winid)
   endfunction
 endif
 
 "
 " move
 "
 if has('nvim')
-  function! s:_move(win, style) abort
-    call nvim_win_set_config(a:win, s:_style(a:style))
+  function! s:_move(self, winid, bufnr, style) abort
+    call nvim_win_set_config(a:winid, s:_style(a:style))
+    if a:bufnr != nvim_win_get_buf(a:winid)
+      call nvim_win_set_buf(a:winid, a:bufnr)
+    endif
+    call s:Window.scroll(a:winid, a:style.topline)
+    return a:winid
   endfunction
 else
-  function! s:_move(win, style) abort
-    call popup_move(a:win, s:_style(a:style))
+  function! s:_move(self, winid, bufnr, style) abort
+    " vim's popup window can't change bufnr so open new popup in here.
+    if a:bufnr != winbufnr(a:winid)
+      let l:On_closed = a:self._on_closed
+      let a:self._on_closed = { -> {} }
+      call s:_close(a:winid)
+      let a:self._on_closed = l:On_closed
+      return s:_open(a:bufnr, a:style, { -> a:self._on_closed() })
```

Comment:
vim's popup does not support bufnr change so we should re-open popup-window (with avoiding `on_close` call...)